lightning/src/util/config.rs

```diff
@@ -161,6 +161,21 @@ pub struct ChannelConfig {
 	///
 	/// Default value: 0.
 	pub fee_proportional_millionths: u32,
+	/// The difference in the CLTV value between incoming HTLCs and an outbound HTLC forwarded over
+	/// the channel this config applies to.
+	///
+	/// This is analogous to [`ChannelHandshakeConfig::our_to_self_delay`] but applies to in-flight
+	/// HTLC balance when a channel appears on-chain whereas
+	/// [`ChannelHandshakeConfig::our_to_self_delay`] applies to the remaining
+	/// (non-HTLC-encumbered) balance.
```

Comment:
The analogy isn't straighforward.

CSV delay protects against the risk of a revoked commitment state and offer to the offended counterparty a delay to punish and claim back funds.

CLTV delta protects against the risk of differing settlement result on your backward/forward channels and offer to a routing node a delay to confirm the commitment and corresponding HTLC transaction before the incoming HTLC can be cancel by the backward counterparty.

What purpose does it serve to document CSV delay around CLTV delta ? It might confuse the read more than anything else, you can set up both of them quite independently. Though ideally both of them should account how much you "trust" your counterparty...

Reply:
Hmm, I'm open to better wording. The intent of the connection here is that both have similar "you must be online at least once per X" requirements, which I think is a useful connection for users. That said, when we handle #850 it becomes more complicated - "you must be online at least once per X, unless you have limited the htlc_in_flight amount to something you don't care about, in which case, we don't care about this".

Reply:
I think your onliness requirement can be reduced to the smaller one between soonest-HTLC expiration and CSV delay. You should care about the smaller one.

I don't think CLTV delta is the same than "soonest-HTLC expiration". What you should look for is the soonest to expire HTLC. If you're block 100, and expiration is 120, your "must-be-back-every X" _dynamically_ decrease until reaching X - CLTV_CLAIM_BUFFER and you must go online.

I think a lightweight onliness requirement is achievable for non-forward node but in that case you don't care about CLTV_DELTA.

Reply:
DIscussed this in a long conversation on IRC, but I think we ultimately agreed.

<details> 
  <summary>IRC Convo Pasted Below</summary>
<pre>
&ltariard&gt BlueMatt: i dunno about your "must-be-back-online-every-X" reasoning around https://github.com/rust-bitcoin/rust-lightning/pull/849#discussion_r597995251]
&ltBlueMatt&gt ariard: correct, but we don't expose the cltv time of all currently-in-flight htlcs to the user
&ltariard&gt or you might allow to be back online every X but at some point you should be back lower than X
&ltariard&gt otherwise you won't be on time for CLTV_CLAIM_BUFFER
&ltBlueMatt&gt and even if we did, it'd be race-y - you have to (a) disable forwarding, then (b) check the cltvs, then (c) shutdown, scheduling when to go back online
&ltariard&gt like if it's block 100 and expiration=120, back every X=7 doesn't work
&ltBlueMatt&gt on the other hand, I think my statement is correct in that as long as you *are* online once every CLTV_DELTA blocks you're (mostly) ok
&ltBlueMatt&gt modulo desire to rbf, of course
&ltariard&gt yeah modulo rbf
&ltBlueMatt&gt I could expland the comment and note that you should be online a ways more often to get rbf time, but I dont think trying to explain the full nuance of htlc timeouts is worth it
&ltariard&gt hmmmm i think it depends when you start to go offline
&ltariard&gt because if you go offline CLTV_DELTA + 10
&ltariard&gt your model work if you assume you go offline as soon as you committed the htlc
&ltariard&gt in practice you might be around for few blocks then disconnect
&ltBlueMatt&gt i think the solution to that is to have a forwarding_disabled flag
&ltBlueMatt&gt which we currently dont have (though you could set infinite fee, I think)
&ltBlueMatt&gt but, again, thats a lot of nuance that I dont think is worth communicating
&ltBlueMatt&gt if users really dont want to be online more often than once a week, they should set cltv_delta and csv_delay to 1 week and move on
&ltariard&gt that's a lot of nuances for now
&ltBlueMatt&gt then they dont have to think about it
&ltariard&gt they should set cltv_delta to a week for sure
&ltBlueMatt&gt if we try to explain how you have to track all your in-flight htlcs and think about when they time-out, then you have a lot of code for....nothing
&ltBlueMatt&gt when they should just turn up the delta and move on
&ltariard&gt i think it's more about mobile-vs-routing-node
&ltariard&gt like won't make sense to check your in-flight HTLCs for a mobile and decide your onliness requirement dynamically
&ltariard&gt awful ux
&ltBlueMatt&gt somewhat, but if you're mobile, you can just jack up the cltv_delta
&ltBlueMatt&gt routing, the cltv_delta docs are correct now, cause you always have htlcs in flight
&ltBlueMatt&gt mobile, you dont care what it is cause you dont want to forward
&ltariard&gt yeah that's my point
&ltBlueMatt&gt right, but I think in *either* case, the response you have to the docs is correct
&ltariard&gt let's go forward with your reasoning
&ltBlueMatt&gt a mobile user reading it would increase the cltv_delta to something big (fine, they dont want to forward, fine outcome) and a routing node would read it and think about it the same as the csv_delay (which is correct)
&ltariard&gt i'm still thinking if you're a big node, better to be always online and spot weird behaviors as soon as you can
&ltariard&gt not exactly, depend risk of realization of each
&ltBlueMatt&gt yea, I mean I can also expand it to talk about rbf if you want
&ltariard&gt account for timevalue
&ltariard&gt fine for me as long as you mention the CLTV_CLAIM_BUFFER/rbf
&ltBlueMatt&gt ok, I'll note rbf
&ltBlueMatt&gt does the current wording not capture CLTV_CLAIM_BUFFER correctly?
&ltBlueMatt&gt ariard: ^?
&ltariard&gt just add a Big Comment on #845
&ltariard&gt let me see
&ltariard&gt BlueMatt: "plus some margin to allow us enough time to broadcast and confirm a transaction" minus some margin ?
&ltBlueMatt&gt yea
&ltBlueMatt&gt well, yes, sorry
&ltariard&gt otherwise looks good to mw
&ltBlueMatt&gt cool, thanks
&ltBlueMatt&gt ariard: can you ack 849?
&ltBlueMatt&gt I split the default/min and tweaked the wording

</pre>
 </details>